rectools/models/nn/transformer_base.py

```diff
@@ -855,3 +890,73 @@ def _get_config(self) -> TransformerModelConfig_T:
         params = {attr: getattr(self, attr) for attr in attrs if attr != "cls"}
         params["cls"] = self.__class__
         return self.config_class(**params)
+
+    def __getstate__(self) -> object:
+        if self.is_fitted:
+            with NamedTemporaryFile() as f:
+                self.fit_trainer.save_checkpoint(f.name)
+                state = torch.load(f.name, weights_only=False)
+                state["is_fitted"] = True
+            return state
+        state = {"model_config": self.get_config(simple_types=True), "is_fitted": False, "trainer": self._trainer}
+        return state
+
+    def __setstate__(self, state: tp.Dict[str, tp.Any]) -> None:
+        if state["is_fitted"] is True:
+            model_config = state["hyper_parameters"]["model_config"]
+            config = self.config_class.model_validate(model_config).model_dump(mode="pydantic")
```

Comment:
Why do you use different ways of working with model config in fitted/non-fitted cases? The logic here is the same, the only difference is the location of the config in the state dict, right?

Not it's confusing

Reply:
fixed